internal/oci/runtime_oci.go

```diff
@@ -846,42 +866,55 @@ func (r *runtimeOCI) StopLoopForContainer(c *Container) {
 				close(done)
 				return
 			}
-			// the PID is still active and belongs to the container, continue to wait
-			time.Sleep(100 * time.Millisecond)
+			// The PID is still active and belongs to the container, continue to wait.
+			time.Sleep(stopProcessWatchSleep)
 		}
 	}()
 
 	// Operate in terms of targetTime, so that we can pause in the middle of the operation
 	// to catch a new timeout (and possibly ignore that new timeout if it's not correct to
 	// take a new one).
-	targetTime := time.Unix(1<<50-1, 0)
-	for finished := false; !finished; {
+	targetTime := time.Now().AddDate(+1, 0, 0) // A year from this one.
+
+	//nolint:staticcheck
+	// The API is due to be deprecated. However, the replacement is completely broken, see:
+	//   https://github.com/kubernetes/kubernetes/issues/118638
+	backoffManager := kwait.NewExponentialBackoffManager(
+		stopInitialBackoff,
+		stopMaximumBackoff,
+		stopResetBackoff,
+		stopBackoffFactor,
+		stopBackoffJitter,
+		&kclock.RealClock{},
+	)
```

Comment:
nit: personal preference to not define the variable if we don't need it. we can inline it in the BackoffUntil function below

Reply:
I thought of doing it, but when done the readibility took a bit of a hit for the worse. The large block is in the middle which makes it somewhat unwildely.

Happy to change it for you. 😀

Reply:
Done.